src/lib/ffi/ffi_tpm2.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+* (C) 2024 Jack Lloyd
+* (C) 2024 René Meusel, Amos Treiber - Rohde & Schwarz Cybersecurity GmbH
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/ffi.h>
+
+#include <botan/internal/ffi_rng.h>
+#include <botan/internal/ffi_util.h>
+
+#if defined(BOTAN_HAS_TPM2)
+   #include <botan/tpm2.h>
+   #include <botan/tpm2_rng.h>
+#endif
+
+extern "C" {
+
+using namespace Botan_FFI;
+
+#if defined(BOTAN_HAS_TPM2)
+/**
+ * This wrapper is required since BOTAN_FFI_DECLARE_STRUCT internally produces a unique pointer,
+ * but the TPM2_Context is meant to be used as a shared pointer.
+ */
+struct botan_tpm2_ctx_wrapper {
+      std::shared_ptr<Botan::TPM2_Context> ctx;
+};
+
+BOTAN_FFI_DECLARE_STRUCT(botan_tpm2_ctx_struct, botan_tpm2_ctx_wrapper, 0xD2B95E15);
+#endif
+
+int botan_tpm2_ctx_init(botan_tpm2_ctx_t* ctx_out, const char* tcti_nameconf) {
+#if defined(BOTAN_HAS_TPM2)
+   return ffi_guard_thunk(__func__, [=]() -> int {
+      if(ctx_out == nullptr) {
+         return BOTAN_FFI_ERROR_NULL_POINTER;
+      }
+      auto ctx = std::make_unique<botan_tpm2_ctx_wrapper>();
+
+      auto tcti = [=]() -> std::optional<std::string> {
```

Comment:
Is the empty string (potentially) meaningful here? Currently we map nullptr to none, and anything else including the empty string to some. Just wondering if we should also map empty to nullopt here.

[Disclaimer I have no idea what this string is doing]

Reply:
Both a nullptr and the empty string indicate the default Transmission Interface. Mapping empty to nullopt here would be functionally equivalent - however in TPM2_Context the none gets translated back to a nullptr given to the TSS so I guess with the current way we keep the translations more consistent.